drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java

```diff
@@ -16,25 +16,34 @@
 
 package org.drools.workbench.screens.guided.dtable.client.widget.table.columns.control;
 
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
 import com.google.gwt.user.client.Command;
 import com.google.gwt.user.client.ui.Anchor;
 import org.drools.workbench.screens.guided.dtable.client.resources.i18n.GuidedDecisionTableConstants;
-import org.uberfire.ext.widgets.common.client.common.popups.YesNoCancelPopup;
+import org.uberfire.ext.editor.commons.client.file.popups.DeletePopUpPresenter;
 
+@Dependent
 public class DeleteColumnManagementAnchorWidget extends Anchor {
 
-    public DeleteColumnManagementAnchorWidget(final String columnHeader,
-                                              final Command afterDeleteConfirmedCommand) {
+    private Command afterDeleteConfirmedCommand;
+
+    @Inject
+    private DeletePopUpPresenter deletePopUpPresenter;
+
+    public DeleteColumnManagementAnchorWidget() {
         super();
         setText(GuidedDecisionTableConstants.INSTANCE.Delete());
         setTitle(GuidedDecisionTableConstants.INSTANCE.DeleteThisColumn());
         addClickHandler((clickEvent) -> {
-            final String confirmMessage = GuidedDecisionTableConstants.INSTANCE.DeleteColumnWarning(columnHeader);
-            YesNoCancelPopup.newYesNoCancelPopup(GuidedDecisionTableConstants.INSTANCE.Delete(),
-                                                 confirmMessage,
-                                                 () -> afterDeleteConfirmedCommand.execute(),
-                                                 null,
-                                                 null).show();
+            deletePopUpPresenter.getToggleCommentPresenter().getViewElement().setAttribute("hidden",
+                                                                                           "true");
+            deletePopUpPresenter.show((comment) -> afterDeleteConfirmedCommand.execute());
```

Comment:
The ```DeletePopupPresenter.View``` shows a "Are you sure you want to delete this **Asset**" prompt. We'd need to be able to make this "Are you sure you want to delete this **column**?". There's probably different ways you might be able to do this; either investigate a ```DeletePopupPresenter.setPrompt(String)``` method (that calls ```View.setPrompt(String)``` that calls ```modal.setTitle( prompt )```) or have your own implementation of ```DeletePopupPresenter``` that injects a different ```View``` that sets the caption to "Are you sure you want to delete this **column**?" but that's a lot of leg work). Up to you.